src/cnd/cnd.ts

```diff
@@ -329,6 +329,218 @@ export interface SwapProperties {
   [k: string]: any;
 }
 
+/**
+ * The payload returned when fetching one swap on the `/swaps/:id` endpoint
+ */
+export interface SwapResponse extends Entity {
+  properties?: FullSwapProperties;
+}
+
+/**
+ * Element of the array in the payload returned when fetching all swaps on the `/swaps/` endpoint
+ */
+export interface SwapElementResponse extends EmbeddedRepresentationSubEntity {
+  properties?: PartialSwapProperties;
+}
+
+/**
+ * The overall status of a swap
+ */
+export enum SwapStatus {
+  /**
+   * The swap was created but the communication phase hasn't yet been finalized
+   */
+  Created = "CREATED",
+  /**
+   * The communication was finalized and blockchain actions are needed or happening
+   */
+  InProgress = "IN_PROGRESS",
+  /**
+   * The swap is finished and the assets were swapped
+   */
+  Swapped = "SWAPPED",
+  /**
+   * The swap is finished and the assets were not swapped
+   */
+  NotSwapped = "NOT_SWAPPED",
+  /**
+   * An unexpected internal failure aborted the swap
+   */
+  InternalFailure = "INTERNAL_FAILURE"
+}
+
+/**
+ * The parameters for one side of the swap
+ */
+export interface Parameters {
+  // Do we want enums instead?
+  /**
+   * Protocol: "han", "herc20" or "halight"
+   */
+  protocol: string;
+  /**
+   * Ledger: "ethereum", "bitcoin" or "lightning"
+   */
+  ledger: string;
+  /**
+   * Network of the ledger: "mainnet", "testnet", "regtest" for Bitcoin, a chain id number for Ethereum
+   */
+  // This is tricky, an alternative would be to have more nesting where each ledger type (Bitcoin, Ethereum, Lightning) have a specific interface to declare network and asset
+  network: string | number;
+  /**
+   * Asset: "bitcoin", "ether" or { standard: "erc20", contract_address: "0x..." } for non-native tokens
+   */
+  // Not sure this is ideal, let's discuss and see what we can come up with
+  asset: string | { standard: "erc20"; contract_address: "0x..." };
+  /**
+   * Expiry: Expiry applying to this ledger, it can be an absolute or relative expiry, depending on the protocol and ledger
+   */
+  // We could have an object that allows the specification of the type of expiry here
+  expiry: number;
+}
+
+export enum EscrowStatus {
```

Comment:
I think this will be useful, at least for the SDK's state machine to know what to do next.

Reply:
Well, clients should never make assumptions about what to do next based on the state :grimacing: 

They should only act on the precense/absence of actions otherwise you are re-implementing the state machine we already have in the backend.

These states should IMO be used purely for presentational purposes :)

Reply:
Yes the idea is for presentational purposes but having a static list allows you the app to show a specific translation/icon/explanation for each status to the user.